src/flint/fq_default.jl

```diff
@@ -675,20 +645,50 @@ function neg!(z::FqFieldElem, a::FqFieldElem)
   return z
```

Comment:
```suggestion
  z.poly = nothing
  return z
```
here as well

Reply:
I have no idea what these `z.poly = nothing` do, but I am happy to trust you if you say they are needed.

Perhaps they then should also go into the new `mul!` methods? I didn't add them there because the existing code didn't do it either. Perhaps it is not needed there? Or perhaps it is needed and the existing code is buggy? (In that case adding some tests that trigger the bug / verify it is fixed would be nice). Unfortunately the `.poly` field is not documented. Some grepping shows a function `_as_poly` but I really don't feel like playing detective to unravel this.

Reply:
Yes, for `mul!`, there is https://github.com/Nemocas/Nemo.jl/pull/1906#discussion_r1806137922.

From what I found: The `.poly` field caches the return value of `_as_poly`. Its only use-case currently is `coeff(x::FqFieldElem, ::Int)` in the case that `parent(x)` is a relative field. `.poly` seems thus seems to contain a representation of `x` as a polynomial over the prime/base field.

In any case, I think it is safer to just clean `.poly` in every mutating operation. Keeping it can lead to wrong results in `coeff`, while deleting it once too often will just re-compute it anew once it is needed the next time.

Reply:
Mutating functions need to either update the `.poly` field or set it to `nothing`.